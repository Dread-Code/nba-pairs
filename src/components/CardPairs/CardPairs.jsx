import React from 'react'
import { Card } from 'semantic-ui-react'
import PropTypes from 'prop-types'

const CardPairs = ({ playersPair }) => (
  <>
    {playersPair.map(player => {
      const header = `${player.first_name} ${player.last_name}`

      return (
        <Card key={header}>
          <Card.Content className="title" header={header} />
          <Card.Content className="description" description={player.h_in} />
        </Card>
      )
    })}
  </>
)

CardPairs.propTypes = {
  playersPair: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default CardPairs
