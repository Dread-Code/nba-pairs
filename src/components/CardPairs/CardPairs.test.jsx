import { shallow } from 'enzyme'
import React from 'react'
import CardPairs from './CardPairs'

const players = [
  {
    first_name: 'Mike',
    h_in: '70',
    h_meters: '1.78',
    last_name: 'Wilks'
  },
  {
    first_name: 'Brevin',
    h_in: '70',
    h_meters: '1.78',
    last_name: 'Knight'
  }
]

describe('CardPairs', () => {
  test('1) Should match snapshoot', () => {
    const wrapper = shallow(<CardPairs playersPair={players} />)
    expect(wrapper).toMatchSnapshot()
  })
  test('2) should find a Deron Williams in the first element', () => {
    const wrapper = shallow(<CardPairs playersPair={players} />)
    expect(wrapper.find('CardContent').getElements()[0].props.header).toBe('Mike Wilks')
    expect(wrapper.find('CardContent').getElements()[1].props.description).toBe('70')
  })
})
