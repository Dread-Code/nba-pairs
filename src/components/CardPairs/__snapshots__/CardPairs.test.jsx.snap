// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPairs 1) Should match snapshoot 1`] = `
<Fragment>
  <Card
    key="Mike Wilks"
  >
    <CardContent
      className="title"
      header="Mike Wilks"
    />
    <CardContent
      className="description"
      description="70"
    />
  </Card>
  <Card
    key="Brevin Knight"
  >
    <CardContent
      className="title"
      header="Brevin Knight"
    />
    <CardContent
      className="description"
      description="70"
    />
  </Card>
</Fragment>
`;
